File: Hrms.Web/src/app/admin/attendance/model/attendance-rule.model.ts
import { SelectListItemModel } from "src/app/shared/models";

export class AttendanceRuleModel {
    id: number;
    companyId: number;
    title: string;
    inTime: string;
    outTime: string;
    firstHalfStart: string;
    firstHalfEnd: string;
    firstDayOfWeek: string;
    lastDayOfWeek: string;
    secondHalfStart: string;
    secondHalfEnd: string;
    graceInTime: string;
    graceOutTime: string;
    totalBreakDuration: string;
    numberOfBreak: number;
    minEffectiveDuration: string;
    autoLeaveDeduction: string;
    minAnomaliesForFirstHalfDeducation: string;
    minAnomaliesForFullyDayDeduction: string;
    startDay: number;
    endDay: number;
    description: string;
    formType: string;
    weekStartDay: SelectListItemModel;
    weekLastDay: SelectListItemModel;
    peopeles : number;
    year:number;
    forwardToNextYear = false;
    constructor() {
        this.weekStartDay = new SelectListItemModel();
        this.weekStartDay = new SelectListItemModel();
    }
}

File: Hrms.Core/Models/Attendance/AttendanceRuleModel.cs
using Hrms.Core.Utilities;

namespace Hrms.Core.Models.Attendance
{
    public class AttendanceRuleModel
    {
        public int Id { get; set; }
        public int CompanyId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public TimeSpan InTime { get; set; }
        public TimeSpan OutTime { get; set; }
        public TimeSpan FirstHalfStart { get; set; }
        public TimeSpan FirstHalfEnd { get; set; }
        public TimeSpan SecondHalfStart { get; set; }
        public TimeSpan SecondHalfEnd { get; set; }
        public TimeSpan GraceInTime { get; set; }
        public TimeSpan GraceOutTime { get; set; }
        public TimeSpan TotalBreakDuration { get; set; }
        public TimeSpan MinEffectiveDuration { get; set; }
        public bool AutoLeaveDeduction { get; set; }
        public int MinAnomaliesForFistHalfDeduction { get; set; }
        public int MinAnomaliesForFullDayDeduction { get; set; }
        public Constants.RecordStatus Status { get; set; }
        public int StartDay { get; set; }
        public int EndDay { get; set; }
        public int NumberOfBreak { get; set; }
        public string FormType { get; set; }
        public int Peoples { get; set; }
        public int Year { get; set; }
        public bool ForwardToNextYear { get; set; }

    }
}
